src/categorical_algebra/CSetDataStructures.jl

```diff
@@ -857,4 +858,75 @@ function init_acset(T::Type{<:ACSet{CD,AD,Ts}},body) where {CD <: CatDesc, AD <:
   code
 end
 
+""" Map over a data type, in the style of Haskell functors
+"""
+
+function Base.map(acs::ACSet; kwargs...)
+  fns = (;kwargs...)
+  _map(acs, fns)
+end
+
+function sortunique!(x)
```

Comment:
It'd be nice if this were built into Julia. I call `unique!(sort!(x))` all over the codebase.